critiquebrainz/frontend/views/test/test_review.py

```diff
@@ -312,3 +320,20 @@ def test_hide_unhide(self):
             follow_redirects=True,
         )
         self.assertIn("Review is not hidden.", str(response.data))
+
+    def test_hide_redirect(self):
```

Comment:
I think that we should split this test into specific tests for each of these two things

* Have a user object (instead of `self.hacker`, for example `self.moderator`) that makes `is_admin` return True
* Have a specific view test that makes a post to /review/x/hide, and then reads the review from the database to check that the hidden flag is True
* Have a separate test that creates a hidden review directly (using create_dummy_review), and then tries to GET to /review/write/x/y and check that the redirect url (and any specific popup messages) are correct.

Reply:
I think the first two parts are already covered in this [test_hide_unhide](https://github.com/metabrainz/critiquebrainz/blob/d9d787bd3b6d30b8f894019a6159d44d00d46482/critiquebrainz/frontend/views/test/test_review.py#L254).

Reply:
I wanted to use `create_dummy_review` but it does not take a `is_hidden` parameter. But that is not much an issue, as there is a `set_hidden_state` so that is easy to fix.

Reply:
This looks much better, thanks